sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/resource/status/StatusCountResourceService.groovy

```diff
@@ -0,0 +1,57 @@
+package life.qbic.portal.sampletracking.resource.status
+
+import life.qbic.datamodel.samples.Status
+import life.qbic.portal.sampletracking.communication.Topic
+import life.qbic.portal.sampletracking.resource.ResourceService
+
+/**
+ * <b>A resource service holding data on status counts per project</b>
+ *
+ * <p>This service offers functionality to publish status counts for a specific project.</p>
+ *
+ * @since 1.0.0
+ */
+class StatusCountResourceService extends ResourceService<StatusCount>{
+
+    StatusCountResourceService() {
+        this.addTopic(Topic.SAMPLE_RECEIVED_COUNT_UPDATE)
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>An {@link IllegalArgumentException} is thrown in case the status is unknown to the service</p>
+     * @param statusCount the status count to add
+     * @throws IllegalArgumentException in case the status is unknown to the service
+     */
+    @Override
+    void addToResource(StatusCount statusCount) throws IllegalArgumentException {
+        switch (statusCount.status) {
+            case Status.SAMPLE_RECEIVED:
+                publish(statusCount, Topic.SAMPLE_RECEIVED_COUNT_UPDATE)
+                content.add(statusCount)
+                break
+            default:
+                throw new IllegalArgumentException("Could not process status count for status: $statusCount.status")
```

Comment:
we do not increase anything here. We set it to a specific value. 
For example when we have a project with 20 samples. When 18 samples are already received and sample 19 is set to the current status bein sample received, we expect the service to be called with `addToResource({projectcode, SAMPLE_RECEIVED, 19})` because we now have 19 samples that are received. We do not call `addToResource({projectcode, SAMPLE_RECEIVED, 1})`